### PR TITLE
Rhmap 13226 - Can't import a previously exported empty json collection within the Studio's Data Browser

### PR DESCRIPTION
**Pull request by rachael-oregan**
## Motivation
A previously exported zipped group of json collections from the Studio contained an empty json collection. Importing this .json collection within the Studio's Data Browser returns an unclear error message.

## Result
Return a message with a list of the collections that were successfully imported.

## Jira
https://issues.jboss.org/browse/RHMAP-13226

### CODE REVIEW

File: lib/ditcher.js [hunk 1]
```diff
@@ -590,6 +590,9 @@ Ditcher.prototype.doImport = function (params, cb) {
     var collectionsToImportArray = Object.keys(collectionsToImport);
 
     collectionsToImportArray.forEach(function(colKey){
+      if (collectionsToImport.colKey == undefined) {
```
Comment (fh-qube) [inline lib/ditcher.js]: BLOCKER Replace "==" with "===". [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AEqEqEq)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Function has a complexity of 22 which is greater than 20 authorized.](https://github.com/feedhenry/fh-db/blob/ffe4e1839d31eac6dda9bd0babbf416a66b2be66/lib/ditcher.js#L573) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Function has a complexity of 22 which is greater than 20 authorized.](https://github.com/feedhenry/fh-db/blob/4653f5147370870e50cb52f253297be65b3da809/lib/ditcher.js#L573) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Function has a complexity of 22 which is greater than 20 authorized.](https://github.com/feedhenry/fh-db/blob/1229d57d01f28d7e8c7801cab88d72ae85b6537b/lib/ditcher.js#L573) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: lib/ditcher.js [hunk 2]
```diff
@@ -590,6 +590,9 @@ Ditcher.prototype.doImport = function (params, cb) {
     var collectionsToImportArray = Object.keys(collectionsToImport);
 
     collectionsToImportArray.forEach(function(colKey){
+      if (collectionsToImport[colKey].length === 0) {
+        return cb(new Error("collection " + colKey + " is empty"));
```
Comment (pb82) [inline lib/ditcher.js]: The outer `´forEach` will still continue after this. What if there is more than one collection and only one of them is empty? Do we want to throw an error or import the non-empty ones?

Comment (rachael-oregan) [inline lib/ditcher.js]: If there is more than one collection and if any of those collections are empty then an error is returned. Should we allow users to upload empty collections? The ticket just mentioned to return a useful error?

Comment (sedroche) [inline lib/ditcher.js]: We are sending an error and aborting execution currently for duplicate key error https://github.com/rachael-oregan/fh-db/blob/1229d57d01f28d7e8c7801cab88d72ae85b6537b/lib/ditcher.js#L610
maybe we can add this check into the importers which will mirror current functionality?

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Function has a complexity of 22 which is greater than 20 authorized.](https://github.com/feedhenry/fh-db/blob/a062f4f6c44520e92b20fa3c259dce50996df856/lib/ditcher.js#L573) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Function has a complexity of 23 which is greater than 20 authorized.](https://github.com/feedhenry/fh-db/blob/6c57133b155a1cddc0ef253486e16317747148a4/lib/ditcher.js#L576) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: lib/ditcher.js [hunk 3]
```diff
@@ -591,11 +591,18 @@ Ditcher.prototype.doImport = function (params, cb) {
...
-      data = collectionsToImport[colKey];
-      importers[colKey] = function(callback){
+    collectionsToImportArray.forEach(function (colKey) {
+      if (collectionsToImport[colKey].length === 0) {
+        emptyCollections.push(colKey);
+      }
+      var name = constructCollectionName({ __fhdb: params.__fhdb, __dbperapp: params.__dbperapp, type: colKey }),
+        data = collectionsToImport[colKey];
+      importers[colKey] = function (callback) {
+        if (collectionsToImport[colKey].length == 0) {
```
Comment (fh-qube) [inline lib/ditcher.js]: BLOCKER Replace "==" with "===". [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AEqEqEq)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Function has a complexity of 23 which is greater than 20 authorized.](https://github.com/feedhenry/fh-db/blob/2f654fd07b320acf9b7bab609e71c4bafa6c9b35/lib/ditcher.js#L576) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Function has a complexity of 23 which is greater than 20 authorized.](https://github.com/feedhenry/fh-db/blob/d9ce6e230e1401082471a0e2f2f9d1773d895795/lib/ditcher.js#L576) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Function has a complexity of 21 which is greater than 20 authorized.](https://github.com/feedhenry/fh-db/blob/bf71a060d65dc5fc067444d218f328ceba8ee489/lib/ditcher.js#L576) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Function has a complexity of 21 which is greater than 20 authorized.](https://github.com/feedhenry/fh-db/blob/f692a8fd80307c77a4b6c55ada8fead4a29c0b9c/lib/ditcher.js#L576) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Pull request merged**